homework1/4var/main.py

```diff
@@ -0,0 +1,97 @@
+# Вариант 4
+# Дана целочисленная квадратная матрица. Определить:
+# 1) произведение элементов в тех строках, которые не содержат отрицательных элементов;
+# 2) максимум среди сумм элементов диагоналей, параллельных главной диагонали матрицы.
+import random
+
+
+def create_random_matrix(num_rows, num_columns):
+    matrix = [[0 for _ in range(num_columns)] for _ in range(num_rows)]
+    for i in range(num_rows):
+        for j in range(num_columns):
+            # matrix[i][j] = random.choice([-1, 1]) * random.randint(1, 100)
+            matrix[i][j] = random.randint(-50, 200)
+    return matrix
+
+
+def multiply_list_numbers(my_list):
+    result = 1
+    for num in my_list:
+        result *= num
+    return result
+
+
+def sum_diagonal_parallel(matrix):
+    diagonal_sum = []
+
+    for j in range(1, rows):
+        value = 0
+        k = 0
+        while j + k < columns and k < rows:
+            value += matrix[k][j + k]
+            # print("\n value: ", value)
```

Comment:
keep code clear.
if you decide provide code for review it should not contain redundant lines and comments